zug/transducer/eager.hpp

```diff
@@ -22,6 +22,56 @@ namespace zug {
 struct eager_tag
 {};
 
+struct eager_t
+{
+    template <typename Algo>
+    struct uncomposed_transducer_t
+    {
+        Algo algo;
+
+        template <typename StepT>
+        constexpr auto operator()(StepT&& step) const
+        {
+            return reducing_func_t<Algo, std::decay_t<StepT>>{algo, step};
+        };
+    };
```

Comment:
This type can maybe also go, replace with `std::bind`

Reply:
https://en.cppreference.com/w/cpp/utility/functional/bind -> bind is only constexpr since c++20...

Reply:
Oh...

Reply:
But wait a second, the definition of `zug::map` uses a lambda at this level exactly. I'm getting a bit lost on what we are trying to achieve and where the lambda can be used or not :sweat_smile:

Reply:
Oh I get it, so the transducer itself is constexrp, but the *application* of it to a reducing function is not. This simplifies everything a lot. You can use `zug::map` for reference. Would that work for you? What is your need exactly?

Reply:
You just wanted to get rid of the multiple definition issues, right? And be able to use `ZUG_INLINE_CONSTEXPR`

Reply:
I personally won't care if sorted and reversed are constexpr or not; using inline in the place is perfectly fine for me as it resolves the multiple definition issue. Just, if the application of eager is not constexpr, sorted and reversed cannot be declared as constexpr either, as they are just applications. So I think it's up to you: if it's okay that we make it `inline` only, then I guess we better do it, as it preserves readability of the code.

Reply:
https://en.cppreference.com/w/cpp/language/inline -> but from here it seems that c++14 does not support inline variables either.

Reply:
@tusooa yes, inline only is fine with me indeed. I am aware that C++14 does not support `inline` variables, that is why we have the macro `ZUG_INLINE_CONSTEXPR` wIth a slightly similar purpose. Maybe @CJBussey can clarify.

Reply:
Applied the suggestions. But I am thinking, if, instead of defining `sorted = eager(sorted_t{})`, maybe we could make it a function that says `sorted(step) = eager(sorted_t{})(step)` which should have the same semantics and bypass the constexpr variable limitations. Which one would be better?

Reply:
`sorted(step)` needs to be a function object (`struct` or lambda) that you can pass by value because that is the transducer itself.